src/main/duke/io/Parser.java

```diff
@@ -0,0 +1,121 @@
+                } else {
+                    try {
+                        int byIndex = Arrays.asList(inputArray).indexOf("/by");
+                        String deadlineDescription = String.join(" ",
+                                Arrays.copyOfRange(inputArray, 1, byIndex));
+                        String dueDate = String.join(" ",
+                                Arrays.copyOfRange(inputArray, byIndex + 1, inputArray.length));
+                        LocalDateTime.parse(dueDate, DateTimeFormatter.ofPattern("yyyy-MM-dd kkmm"));
+                        // check if the date and time input is in the right format
+                        if (deadlineDescription.equals("") || dueDate.equals("")) {
+                            throw new DukeException("Please specify the description/due date of the deadline task.");
+                        }
+                        newCommand = new CDeadline(deadlineDescription, dueDate);
+                    } catch (DateTimeParseException e){
+                        throw new DukeException("Please provide the date time in this format YYYY-MM-DD 0000");
+                    }
+                }
+                break;
+            case "event":
+                if (!userInput.contains("/at")) {
+                    throw new DukeException("Please specify the date time using the /at keyword.");
+                } else {
+                    try {
+                        int byIndex = Arrays.asList(inputArray).indexOf("/at");
+                        String eventDescription = String.join(" ",
+                                Arrays.copyOfRange(inputArray, 1, byIndex));
+                        String dateTime = String.join(" ",
+                                Arrays.copyOfRange(inputArray, byIndex + 1, inputArray.length));
+                        LocalDateTime.parse(dateTime, DateTimeFormatter.ofPattern("yyyy-MM-dd kkmm"));
+                        // check if the date and time input is in the right format
+                        if (eventDescription.equals("") || dateTime.equals("")) {
+                            throw new DukeException("Please specify the description/date time of the event task.");
+                        }
+                        newCommand = new CEvent(eventDescription, dateTime);
+                    } catch (DateTimeParseException e){
+                        throw new DukeException("Please provide the date time in this format YYYY-MM-DD 0000");
+                    }
+                }
+                break;
+            case "find":
+                String findString = String.join(" ",
+                        Arrays.copyOfRange(inputArray, 1, inputArray.length));
+                newCommand = new CFind(findString);
+                break;
+            default:
+                throw new DukeException("Sorry. I do not understand your input.");
+        }
+        return newCommand;
```

Comment:
I think there is no indentation for _case_ clauses. I suggest you change the setting in the IDE.